submissions/Zakkarat/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,121 @@
+const Net = require("net");
+
+const help = `
+This script used for finding open ports for specified host.
+
+Paramaters: 
+    --host     Paramater which specifies adress, where ports will be searched.
+               Adress must be specified after the parameter.
+               It can be specified as IP adress or URL.
+               Examples: --host google.com,
+               --host 172.217.3.110
+    
+    --port     Parameter which specifies port range, which will be used to find the open ports.
+               After the parameter two numbers must be specified with a delimiter '-' between them.
+               Numbers must be more than 0 and less than 65536, as there are only 2^(16) ports available.
+               Examples: --port 30-562,
+               --port 1-65535,
+               --port 10-99
+
+    --help     Parameter will show this instructions. Hope they were useful!
+`
+
+const toObject = args => {
+  const obj = {};
+  args.forEach((elem, i) => {
+    if (elem === '--help') {
+        console.log(help);
+        obj.help = true;
+    }else if (i % 2 === 0) {
+      elem = elem.slice(2);
+      obj[elem] = args[i + 1];
+    }
+  });
+  return obj;
+};
+
+const checkHost = async host => {
+  if(!host) {
+    console.log('No host specified. Use --help for more information.')
+  }
+  const dns = require("dns").promises;
```

Comment:
I don't see the reason of such in-block dynamic `require` call, since it will be called in 90% of use cases.
Also it might be interesting for you to read this
https://nodesecroadmap.fyi/chapter-2/dynamism.html